### PR TITLE
projects: max14916 : Add project files for MAX14916

### PR DESCRIPTION
**Pull request by RaduSabau1**
## Pull Request Description

Add initial header and source files for MAX14916.
This project was tested on the MAX32690 and MAX32666 microcontrollers and used the max149x6-base driver,
max14916 driver and max14916 iio driver as well.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [ ] I have performed a self-review of the changes
- [ ] I have commented my code, at least hard-to-understand parts
- [ ] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [ ] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: projects/max14916/src/examples/dummy/dummy_example.c [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	enum max14916_sht_vdd_thr sht_vdd_thr = MAX14916_SHT_VDD_THR_14V;
+
+	/* Intializing the device MAX14916. */
+	ret = max14916_init(&max14916_desc, &max14916_ip);
+	if(ret)
+		goto error;
+
+	/* Create a loop that will turn SLEDs on and off like in a "loading screen"
+	   sequence. */
+	while (j < 5) {
```
Comment (rbolboac) [inline projects/max14916/src/examples/dummy/dummy_example.c:63]: uninitialized value, cpp check is complaining about this

File: projects/max14916/src/examples/dummy/dummy_example.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+*******************************************************************************/
+#include "dummy_example.h"
+#include "common_data.h"
+#include "max149x6-base.h"
+#include "max14916.h"
+#include "no_os_delay.h"
+#include "no_os_print_log.h"
+#include "no_os_irq.h"
+
+int dummy_example_main()
```
Comment (rbolboac) [inline projects/max14916/src/examples/dummy/dummy_example.c:47]: would be nice to have a short description about what this example is doing

File: projects/max14916/src/platform/maxim/main.c [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+#ifdef DUMMY_EXAMPLE
+	struct no_os_uart_desc *uart_desc;
+
+	ret = no_os_uart_init(&uart_desc, &max14916_uart_ip);
+	if (ret)
+		return ret;
+
+	no_os_uart_stdio(uart_desc);
+
+	ret = dummy_example_main();
```
Comment (rbolboac) [inline projects/max14916/src/platform/maxim/main.c:68]: missing no_os_uart_remove in case of an error

**Comment by RaduSabau1** [comment]
V2

- Removing uart in case of an error.
- Addressed other comments.

File: projects/max14916/Makefile [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#Select the example you want to enable by choosing y for enabling and n for disabling
+DUMMY_EXAMPLE = y
```
Comment (buha) [inline projects/max14916/Makefile:2]: same comment here as for your other PR, rename `dummy` to `basic` or something else

File: projects/max14916/src/platform/maxim/main.c [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+	if (ret)
+		no_os_uart_remove(uart_desc);
+#endif
+
+#if (DUMMY_EXAMPLE + IIO_EXAMPLE + IIO_TRIGGER_EXAMPLE == 0)
```
Comment (buha) [inline projects/max14916/src/platform/maxim/main.c:73]: You don't have an IIO_TRIGGER_EXAMPLE, probably a bad copy paste

**Comment by RaduSabau1** [comment]
V3

- Switched name of dummy example to basic example
- Addressed other comments.

**Review by amiclaus — APPROVED** [review]

File: projects/max14916/src/examples/basic/basic_example.c [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+
+	/* Reading Config 2 to check if the mentioned above values have been set
+	   correctly. */
+	ret = max149x6_reg_read(max14916_desc, MAX14916_CONFIG2_REG, &reg_val);
+	if (ret)
+		goto error;
+
+	pr_info("CONFIG2 = 0x%x\n", reg_val);
+
+	return 0;
```
Comment (rbolboac) [inline projects/max14916/src/examples/basic/basic_example.c:124]: let's apply the same logic as in the other PRs... before returning 0 remove should be called

File: projects/max14916/src/examples/iio_example/iio_example.c [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+
+	app_init_param.devices = iio_devices;
+	app_init_param.nb_devices = NO_OS_ARRAY_SIZE(iio_devices);
+	app_init_param.uart_init_params = max14916_uart_ip;
+
+	ret = iio_app_init(&app, app_init_param);
+	if (ret)
+		goto app_error;
+
+	return iio_app_run(app);
```
Comment (rbolboac) [inline projects/max14916/src/examples/iio_example/iio_example.c:81]: check return and if an error occurs call remove

Comment (RaduSabau1) [reply projects/max14916/src/examples/iio_example/iio_example.c:81]: my bad, i forgot to push.

**Comment by RaduSabau1** [comment]
V4

- Solved memory leaks.

File: projects/max14916/src/examples/basic/basic_example.c [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+
+	/* Reading Config 2 to check if the mentioned above values have been set
+	   correctly. */
+	ret = max149x6_reg_read(max14916_desc, MAX14916_CONFIG2_REG, &reg_val);
+	if (ret)
+		goto error;
+
+	pr_info("CONFIG2 = 0x%x\n", reg_val);
+
+	max14916_remove(max14916_desc);
```
Comment (buha) [inline projects/max14916/src/examples/basic/basic_example.c:124]: yeah so remove line 124-125

File: projects/max14916/src/examples/basic/basic_example.c [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+	return 0;
+
+error:
+	max14916_remove(max14916_desc);
+init_error:
+	pr_info("Error!\n");
```
Comment (buha) [inline projects/max14916/src/examples/basic/basic_example.c:130]: and put a conditional here, and also print the value of ret:
```
if (ret)
    pr_info("End of program (%d)", ret);
```
something like that

Comment (buha) [reply projects/max14916/src/examples/basic/basic_example.c:130]: of course you can remove the `if (ret)` if you want to see the message when the program ends gracefully

**Comment by RaduSabau1** [comment]
V5

- Renamed error labels to be more intuitive.

**Comment by RaduSabau1** [comment]
V6

- Modified error labels.

**Comment by RaduSabau1** [comment]
V7

- Renamed error labels.

**Comment by RaduSabau1** [comment]
V8

- Fixed warnings related from the commited files.

**Comment by RaduSabau1** [comment]
V9

- Calling no_os_uart_remove after basic_example_main.

**Review by buha — APPROVED** [review]

**Review by rbolboac — APPROVED** [review]

**Pull request merged**